### DISCUSSION TITLE
Almost all devices showing up in obsolete delete LWTs list only

### DISCUSSION THREAD

**Discussion by jmorris644** [jziolkowski/tdm#248] [Q&A] [answered]
I installed a new managed switch and all of my devices except 1 do not show up anymore. (I mention the switch but I am not sure that is the culprit.) But they do show up in the obsolete LTW list. I have tried clearing that list and verified that it is cleared, but the next time I connect to the broker the list gets repopulated.

I cannot see anything special about the single device that DOES appear.

**Comment by barbudor** [comment]
How did you cleared?
How did you checked it was cleared?
Did you used a mqtt debug tool? Which one? How? 

I don't understand what you mean by "the next time I connect to the broker the list gets repopulated"
When you connect what? TDM?

TDM needs the retained LWT topics to know about the devices. If you clear them from your broker, TDM can't know about the devices until a restart of the device.

If you haven't changed your devices fulltopic, if your devices are properly connected to the broker, if TDM can connect to the broker, then everything should work

Without much details on your MQTT configuration, it's hard to guess what goes wrong

**Comment by jmorris644** [comment]
I am happy to provide as much details as possible.

There is an option under the MQTT menu item to view obsolete retained LWTs. I selected All and then hit the delete key to clear them. Did that actually clear them from the broker? Oh, haha. Yes it did. That is why when I opened up MQTT-Explorer they all had to repopulate.

MQTT-Explorer sees everything just fine. No issues.

I preface all of my devices with text. For example, all of my lights begin with the word lights, and then the %prefix%/%topic%/

**Reply by barbudor** [reply]
You must tell TDM about all your different FullTopic that are different from the default

**Comment by jmorris644** [comment, marked as answer]
Interesting... I have never done that before and it has always worked. Is that new with this version of tdm? Or maybe something changed in the newest version of Tas?

I will go ahead and add my prefixes  in the autodiscovery patterns dialog and see what happens.

edit: That did the trick. I am not sure what changed but it is working now. Thanks!

**Reply by barbudor** [reply]
AFAIK It has always been like that
Without that, TDM doesn't know the structure of the topics to subscribe too

But not using custom fulltopics myself I can't confirm how it was before

**Comment by jziolkowski** [comment]
As of today, TDM uses wildcard fulltopics, f.e. `+/%prefix%/%topic%/`, go give it a try if still applicable
